projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -843,17 +845,28 @@ function ( $alert ) {
 	/**
 	 * Collect all possible alerts that we might use a red bubble notification for
 	 *
+	 * @param bool $bypass_cache - whether to bypass the red bubble cache.
 	 * @return array
 	 */
-	public static function get_red_bubble_alerts() {
+	public static function get_red_bubble_alerts( bool $bypass_cache = false ) {
 		static $red_bubble_alerts = array();
 
 		// using a static cache since we call this function more than once in the class
 		if ( ! empty( $red_bubble_alerts ) ) {
 			return $red_bubble_alerts;
 		}
+
+		// check for stored alerts
+		$stored_alerts = get_transient( self::MY_JETPACK_RED_BUBBLE_TRANSIENT_KEY );
+		// Cache bypass for red bubbles should only happen on the My Jetpack page
+		if ( $stored_alerts !== false && ! ( $bypass_cache ) ) {
```

Comment:
This is a nitpick but are the parenthesis around `$bypass_cache` necessary?

Reply:
probably not, I chose that syntax since it was recommended by the phan static code analysis 😅